flycheck.el

```diff
@@ -6372,7 +6376,47 @@ See URL `https://developers.google.com/closure/utilities'."
   :error-patterns ((warning
                     line-start (file-name) ":" line ":("
                     (id (one-or-more digit)) ") " (message) line-end))
-  :modes (js-mode js2-mode js3-mode))
+  :modes (js-mode js2-mode js3-mode)
+  :next-checkers ((warning . javascript-jscs)))
+
+(flycheck-def-config-file-var flycheck-jscsrc javascript-jscs ".jscsrc"
+  :safe #'stringp
+  :package-version '(flycheck . "0.24"))
+
+(defun flycheck-locate-jscs-config-file ()
+  "Locate the config file of JSCS from the current directory.
+
+Return the config file name, or nil, if the config file is not found
+
+See URL `http://jscs.info/overview.html#-config-c-'."
+  (let (file)
+    (locate-dominating-file
+     default-directory
+     (lambda (dir)
+       (or (and (file-exists-p (setq file (expand-file-name "package.json" dir)))
+                (assq 'jscsConfig (json-read-file file)))
+           (file-exists-p (setq file (expand-file-name ".jscsrc" dir)))
+           (file-exists-p (setq file (expand-file-name ".jscs.json" dir)))
+           (setq file nil))))
+    file))
```

Comment:
JSCS works without --config option, but it fails with exit code 1 when it cannot find a config file by itself.